File: web/packages/lib/src/models/scoreboard.ts
import { z } from "@climblive/lib/utils";
import type { ScoreboardEntry } from "./generated";
import { scoreSchema } from "./score";

export const scoreboardEntrySchema: z.ZodType<ScoreboardEntry> = z.object({
  contenderId: z.number(),
  compClassId: z.number(),
  name: z.string(),
  withdrawnFromFinals: z.boolean(),
  disqualified: z.boolean(),
  scrubbedAt: z.coerce.date().optional(),
  score: scoreSchema.optional(),
});

File: backend/internal/domain/scoreboard.go
package domain

type ScoreboardEntry struct {
	ContenderID         ContenderID `json:"contenderId"`
	CompClassID         CompClassID `json:"compClassId"`
	PublicName          string      `json:"publicName,omitempty"`
	ClubName            string      `json:"clubName,omitempty"`
	WithdrawnFromFinals bool        `json:"withdrawnFromFinals"`
	Disqualified        bool        `json:"disqualified"`
	Score               *Score      `json:"score,omitempty"`
}
